src/neptune_pytorch/impl/__init__.py

```diff
@@ -13,25 +13,81 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-__all__ = ["__version__", "NeptuneCallback"]
+__all__ = ["__version__", "NeptuneLogger"]
 
-from typing import Optional
+import os
+import warnings
+import weakref
+from typing import (
+    Optional,
+    Union,
+)
 
 import torch
 
 from neptune_pytorch.impl.version import __version__
 
+try:
+    # neptune-client>=1.0.0 package structure
+    from neptune import Run
+    from neptune.handler import Handler
+    from neptune.internal.utils import verify_type
+    from neptune.types import File
+except ImportError:
+    from neptune.new import Run
+    from neptune.new.handler import Handler
+    from neptune.new.integrations.utils import verify_type
+    from neptune.new.types import File
+
 IS_TORCHVIZ_AVAILABLE = True
 try:
     import torchviz
 except ImportError:
     IS_TORCHVIZ_AVAILABLE = False
 
+INTEGRATION_VERSION_KEY = "source_code/integrations/neptune-pytorch"
+
 
 class NeptuneLogger:
+    """Captures model training metadata and logs them to Neptune.
+
+    Args:
+        run: Neptune run object. You can also pass a namespace handler object;
+            for example, run["test"], in which case all metadata is logged under
+            the "test" namespace inside the run.
+        base_namespace: Namespace where all metadata logged by the callback is stored.
+        log_gradients: Whether to track the frobenius-order norm of the gradients.
+        log_parameters: Whether to track the frobenius-order norm of the parameters.
+        log_freq: How often to log the parameters/gradients norm. Applicable only
+            if `log_parameters` or `log_gradients` is True.
+        log_model_diagram: Whether to save the model visualization.
+            Requires torchviz to be installed: https://pypi.org/project/torchviz/
+
+    Example:
+        import neptune
+        from neptune.integrations.pytorch import NeptuneLogger
+        run = neptune.init_run()
+        neptune_callback = NeptuneLogger(run=run, model=model)
```

Comment:
Just noticed one thing. What's the `model` parameter here?

Reply:
Ah, this is the PyTorch `model` (of type torch.nn.Module) that will be tracked by the integration. Missed that 😓 . Will update the PR soon. 

Thank you!

Reply:
Ping @normandy7

Reply:
Looks good, thanks!